### ISSUE TITLE
When subscribing to newsletter, if email already exists, don't throw exception, but just warn without exception

### ISSUE THREAD

**Issue by woutersamaey** [magento/magento2#19404]
I want to replace the line in vendor/magento/module-newsletter/Controller/Subscriber/NewAction.php
```
throw new \Magento\Framework\Exception\LocalizedException(
     __('This email address is already subscribed.')
);
```
With:
```
$this->messageManager->addErrorMessage(__('This email address is already subscribed.'));
```
Reason is: By throwing an exception, it is later caught, added as error message + logged. Since this is a user error, it should not be logged. Our log files don't need these kinds of messages.

Changing this 1 line solves that and only that.

So, before I make a PR and get into a discussion about this. Is it something you would agree to?

### Preconditions (*)
Latest Magento version

### Steps to reproduce (*)
Try to subscribe an email that is already subscribed. The error log receives a message, but in my view the error log should not get this.

### Expected result (*)
Show error, but don't log, it's a user issue, not a platform issue.

### Actual result (*)
Error message is shown (good) + message added to log (not desired)

I am willing to make a PR, but only if this reasoning is accepted / agreed on.

**Comment by magento-engcom-team** [comment]
Hi @woutersamaey. Thank you for your report. 
To help us process this issue please make sure that you provided the following information:
- [x] Summary of the issue
- [x] Information on your environment
- [x] Steps to reproduce
- [x] Expected and actual results

Please make sure that the issue is reproducible on the vanilla Magento instance following **Steps to reproduce**. To deploy vanilla Magento instance on our environment, please, add a comment to the issue: 

`@magento-engcom-team give me $VERSION instance` 

where `$VERSION` is [version tags](https://github.com/magento/magento2/tags) (starting from 2.2.0+) or develop branches (for example: 2.3-develop). 
For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation.

@woutersamaey do you confirm that you was able to reproduce the issue on vanilla Magento instance following steps to reproduce?
- [x] yes
- [ ] no

**Comment by magento-engcom-team** [comment]
Hi @engcom-backlog-nazar. Thank you for working on this issue. 
In order to make sure that issue has enough information and ready for development, please read and check the following instruction: :point_down:
- [x] 1. Verify that issue has all the required information. (**Preconditions**, **Steps to reproduce**, **Expected result**, **Actual result**).<details><summary>Details</summary>If the issue has a valid description, the label `Issue: Format is valid` will be added to the issue automatically. Please, edit issue description if needed, until label `Issue: Format is valid` appears.</details>

- [x] 2. Verify that issue has a meaningful description and provides enough information to reproduce the issue. If the report is valid, add `Issue: Clear Description` label to the issue by yourself.

- [x] 3. Add `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.
- [x] 4. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [x] 5. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [x] 6. Add label `Issue: Confirmed` once verification is complete. 

- [x] 7. Make sure that automatic system confirms that report has been added to the backlog.

**Comment by magento-engcom-team** [comment]
@engcom-backlog-nazar Thank you for verifying the issue. Based on the provided information internal tickets `MAGETWO-96626`, `MAGETWO-96627` were created

**Comment by orlangur** [comment]
@woutersamaey for simple code changes, like this one, please propose a PR directly. Thanks for collaboration!

**Comment by woutersamaey** [comment]
@orlangur will you accept this change or not?

**Comment by orlangur** [comment]
@woutersamaey surely,

> Since this is a user error, it should not be logged. Our log files don't need these kinds of messages.

was enough to convince me :)

**Comment by woutersamaey** [comment]
@orlangur  Ok :-) I'll make a PR then. Just wanted to check first before going though the motions.
If there is a better way to get feedback on small questions, I'm happy to go about it another way too. What is convenient in these cases?

**Comment by woutersamaey** [comment]
@orlangur see https://github.com/magento/magento2/pull/19406

**Comment by orlangur** [comment]
@woutersamaey the main is not wasting efforts I believe, when you can change code quickly (even if it doesn't really work on first attempt but shows an idea) it is better to post a PR, otherwise it can be discussed in Slack or reported as an issue.

**Comment by woutersamaey** [comment]
@orlangur Got it. How do I get access to https://magentocommeng.slack.com ? Looks like I need an account.

**Comment by orlangur** [comment]
@woutersamaey try https://tinyurl.com/engcom-slack

**Comment by magento-engcom-team** [comment]
Hi @Dharmeshvaja91. Thank you for working on this issue. 
Looks like this issue is already verified and confirmed. But if your want to validate it one more time, please, go though the following instruction:
- [ ] 1. Add/Edit `Component: XXXXX` label(s) to the ticket, indicating the components it may be related to.

- [ ] 2. Verify that the issue is reproducible on `2.3-develop` branch<details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.3-develop instance` to deploy test instance on Magento infrastructure. <br/> - If the issue is reproducible on `2.3-develop` branch, please, add the label `Reproduced on 2.3.x`.<br/>- If the issue is not reproducible, add your comment that issue is not reproducible and close the issue and **_stop verification process here_**!
</details>

- [ ] 3. Verify that the issue is reproducible on `2.2-develop` branch. <details><summary>Details</summary>- Add the comment `@magento-engcom-team give me 2.2-develop instance` to deploy test instance on Magento infrastructure. <br/>- If the issue is reproducible on `2.2-develop` branch, please add the label `Reproduced on 2.2.x`</details>

- [ ] 4. If the issue is not relevant or is not reproducible any more, feel free to close it.

**Comment by Dharmeshvaja91** [comment]
@magento-engcom-team give me $VERSION instance

**Comment by magento-engcom-team** [comment]
Hi @woutersamaey. Thank you for your report. 
The issue has been fixed in magento/magento2#19416 by @Dharmeshvaja91 in 2.3-develop branch
Related commit(s):
 - [c3a64f20b9b71d97c0a050a07b02b0aeaad45c26](https://github.com/magento/magento2/commit/c3a64f20b9b71d97c0a050a07b02b0aeaad45c26)
 - [c0d34804b2e3a866dccb8bde4f57a9eafdbf65e4](https://github.com/magento/magento2/commit/c0d34804b2e3a866dccb8bde4f57a9eafdbf65e4)
 - [ca55821d94f9f4b34acc1878f0b9d7cc9a768c27](https://github.com/magento/magento2/commit/ca55821d94f9f4b34acc1878f0b9d7cc9a768c27)
 - [46b34797d20072e34e0cc87adcbc2cc36d49f281](https://github.com/magento/magento2/commit/46b34797d20072e34e0cc87adcbc2cc36d49f281)

The fix will be available with the upcoming 2.3.1 release.
